### DISCUSSION TITLE
Parameters of `CSXCAD.ContinuousStructure.AddDump` ?

### DISCUSSION THREAD

**Discussion by SengerM** [thliebig/openEMS-Project#248] [Q&A Usage]
I want to know about the parameters of [`CSXCAD.ContinuousStructure.AddDump`](https://docs.openems.de/python/CSXCAD/CSXCAD.html#CSXCAD.ContinuousStructure.AddDump). I found some information in [`CSXCAD.CSProperties.CSPropDumpBox`](https://docs.openems.de/python/CSXCAD/CSProperties/CSPropDumpBox.html#CSXCAD.CSProperties.CSPropDumpBox) and in [the analogous function from the matlab interface](https://github.com/thliebig/CSXCAD/blob/9257bf17b6bbce85ea00d9f85b89416530563cd2/matlab/AddDump.m), but I see it has more arguments.

- Are the arguments the same in Python?
- Is there more information available on the meaning of arguments like `SubSampling`, `OptResolution`, etc? I don't understand the notation `%   SubSampling:   field domain sub-sampling, e.g. '2,2,4'`, what is the meaning of `2,2,4` ?

**Comment by Juleinn** [comment]
About the arguments, I don't know but I tend to define the individual properties with the dedicated functions, like this : 
```python
dumpbox = CSX.AddDump(filename_prefix)
dumpbox.SetDumpType(0)  #E-field 
dumpbox.SetDumpMode(0) #no interpolation
dumpbox.AddBox(start, stop)
```

For Subsampling and OptResolution, based on the legacy wiki entry https://wiki.openems.de/index.php/Dump_Box_Property.html
You can reduce the spatial density of your dumps (I assume to reduce file size, because it is quite easy to generate 50+GB of dump data in one sim). 
SetSubSampling(2,2,4) -> dump every 2nd value on x and y, every 4th value on z
SetOptResolution(2,2,4) -> dump such that there is a value on ~every 2drawing unit for x and y, ~every 4th drawing unit for z

You'd probably use only one of the two functions, I don't know which takes precedence over the other if you use both
